transcom/mymove
# INT B-21657 - Prime request service items ppm

## [B-21657](https://www13.v1host.com/USTRANSCOM38/story.mvc/summary?oidToken=Story%3A1029733)

## Summary

This PR introduces another validation layer to the Prime `operationId: createMTOServiceItem` API (version 1) to exclude operation for PPM shipments.  Service response will return 422 code indicating shipment type PPM is not allowable.

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Note: **`operationId: createMTOServiceItem` API  is only at version 1.** Changes were also applied to version 2 and version 3 handlers to ensure the changes are cascaded to all potential API versions.

1. Login as customer and onboard/submit a move containing both HHG and PPM. For each use postal code **62226** for all pickup addresses and **02542** for all destination addresses. This is to ensure the entire workflow will be associated to the AGFM GBLOC.
2. Login as New Service Counselor for AGFM. Find move and submit move details.
3. Login as TOO for AGFM. Select HHG and Approve selected to add service items.
4. Login as Prime Sim for AGFM. Find move and Add Service Item for PPM. Verify ALL service item types are prevented from executing. Make sure to open web developer tool and capture JSON payload. Service Response must return 422 --  Error: Unprocessable Entity.
![Screenshot 2024-12-17 at 11 07 07 AM](https://github.com/user-attachments/assets/5a553e33-618b-4a98-a65c-5d62f857f130)
5. Goto http://primelocal:3000/swagger-ui/prime.html#/mtoServiceItem/createMTOServiceItem . Copy and paste JSON payload from step 4 and execute API. Verify response is 422:
 ![Screenshot 2024-12-17 at 11 08 04 AM](https://github.com/user-attachments/assets/113fd621-e8ab-4c56-adff-9ba8cef60835)
6. Verify shipment types NOT PPM can still create service items from Prime via Prime Sim, Swagger, Postman ..etc...


 

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
